### PR TITLE
Remove polling from addon manager

### LINKED ISSUE #6034: Remove polling from the addon manager to reduce overhead
In https://github.com/kubernetes/minikube/issues/5682#issuecomment-555285979, I pointed out that the addon manager polls to make sure addon configuration is correct every 5 seconds. This can result in CPU spikes up to ~80% of a core every 5 seconds.

Every 5 seconds, the addon manager runs the same command:

```
/usr/local/bin/kubectl apply -f /etc/kubernetes/addons -l kubernetes.io/cluster-service!=true,addonmanager.kubernetes.io/mode=Reconcile --prune=true --prune-whitelist core/v1/ConfigMap --prune-whitelist core/v1/Endpoints --prune-whitelist core/v1/Namespace --prune-whitelist core/v1/PersistentVolumeClaim --prune-whitelist core/v1/PersistentVolume --prune-whitelist core/v1/Pod --prune-whitelist core/v1/ReplicationController --prune-whitelist core/v1/Secret --prune-whitelist core/v1/Service --prune-whitelist batch/v1/Job --prune-whitelist batch/v1beta1/CronJob --prune-whitelist apps/v1/DaemonSet --prune-whitelist apps/v1/Deployment --prune-whitelist apps/v1/ReplicaSet --prune-whitelist apps/v1/StatefulSet --prune-whitelist extensions/v1beta1/Ingress --recursive 
```

Instead of running the addon manager as a pod by default, we can turn the addon manager addon off by default and run the `kubectl apply` command ourselves whenever a user changes the state of addons via `minikube addons enable/disable`

### PR DESCRIPTION
**Pull request by priyawadhwa**
This PR removes polling from the addon manager to reduce CPU overhead. 

Fixes #6034

### CODE REVIEW

File: pkg/minikube/addons/reconcile.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+func ReconcileAddons(cmd command.Runner) error {
+	if _, err := cmd.RunCmd(kubectlCommand()); err != nil {
+		glog.Warningf("reconciling addons failed: %v", err)
+		return err
+	}
+	return nil
+}
+
+func kubectlCommand() *exec.Cmd {
+	kubectlBinary := fmt.Sprintf("/var/lib/minikube/binaries/%s/%s", constants.KubectlBinaryVersion, constants.KubectlBinary)
```
Comment (medyagh) [inline pkg/minikube/addons/reconcile.go]: just a knit but could use a path Join here

File: pkg/minikube/constants/constants.go [hunk 2]
```diff
@@ -87,6 +87,14 @@ var ImageRepositories = map[string][]string{
 // KubeadmBinaries are Kubernetes release binaries required for kubeadm
 var KubeadmBinaries = []string{"kubelet", "kubeadm"}
 
+const (
+	// KubectlBinary is the name of the kubectl binary installed in minikube
+	KubectlBinary = "kubectl"
+	// KubectlBinaryVersion is the version of kubectl installed in minikube
+	// it's set to v1.13.2 to match the kubectl version in the addon manager image
+	KubectlBinaryVersion = "v1.13.2"
```
Comment (medyagh) [inline pkg/minikube/constants/constants.go]: if this is only used by the addon package, lets move it there

File: pkg/minikube/addons/reconcile.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+	args := []string{"KUBECONFIG=/var/lib/minikube/kubeconfig", kubectlBinary, "apply", "-f", "/etc/kubernetes/addons", "-l", "kubernetes.io/cluster-service!=true,addonmanager.kubernetes.io/mode=Reconcile", "--prune=true"}
```
Comment (medyagh) [inline pkg/minikube/addons/reconcile.go]: it is not clear for me what the prune is for, do you mind adding a comment explaining what it is doing?

**Comment by medyagh** [comment]
KVM integration logs 
```
    --- FAIL: TestFunctional/parallel (0.00s)
        --- FAIL: TestFunctional/parallel/AddonManager (301.18s)
            functional_test.go:174: (dbg) TestFunctional/parallel/AddonManager: waiting 5m0s for pods matching "component=kube-addon-manager" in namespace "kube-system" ...
            functional_test.go:174: ***** TestFunctional/parallel/AddonManager: pod "component=kube-addon-manager" failed to start within 5m0s: timed out waiting for the condition ****
            helpers.go:313: TestFunctional/parallel/AddonManager: showing logs for failed pods as of 2019-12-10 21:36:58.916534172 +0000 UTC m=+1206.295678758
            functional_test.go:175: wait: component=kube-addon-manager within 5m0s: timed out waiting for the condition
```

File: pkg/minikube/addons/reconcile.go (hunk 3 above)
Comment (priyawadhwa) [reply pkg/minikube/addons/reconcile.go]: done!

File: pkg/minikube/assets/addons.go [hunk 4]
```diff
@@ -75,7 +75,7 @@ var Addons = map[string]*Addon{
 			"addon-manager.yaml.tmpl",
 			"0640",
 			true),
-	}, true, "addon-manager"),
+	}, false, "addon-manager"),
```
Comment (tstromberg) [inline pkg/minikube/assets/addons.go]: Please follow this up with a PR that removes the addon-manager. I think it's continued existence will only add confusion. 

If someone begs for the addon-manager to  come back, we can reconsider.

**Comment by priyawadhwa** [comment]
HyperV Windows [logs](https://a646c87040050000000000000000001.proxy.googleprod.com/job/Hyper-V_Windows_integration/1623/console)


Hyperkit MacOS [logs](https://a646c87040050000000000000000001.proxy.googleprod.com/job/HyperKit_macOS_integration/6376/consoleFull)

VirtualBox Linux [logs](https://a646c87040050000000000000000001.proxy.googleprod.com/job/VirtualBox_Linux_integration/7753/)

VirtualBox MacOS [logs](https://a646c87040050000000000000000001.proxy.googleprod.com/job/VirtualBox_macOS_integration/8078/)

None [logs](https://a646c87040050000000000000000001.proxy.googleprod.com/job/none_Linux_integration/6851/)

File: pkg/minikube/assets/addons.go (hunk 4 above)
Comment (priyawadhwa) [reply pkg/minikube/assets/addons.go]: sounds good

**Pull request merged**